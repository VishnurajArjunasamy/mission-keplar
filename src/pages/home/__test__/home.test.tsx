import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Home from "../home";

jest.mock("../../../containers/side-bar/side-bar", () =>
  jest.fn(() => <div>Mocked SideBar</div>)
);
jest.mock("../../../containers/blog-list/blog-list", () =>
  jest.fn(() => <div>Mocked BlogList</div>)
);
jest.mock("../../../containers/blog-details/blog-details", () =>
  jest.fn(() => <div>Mocked BlogDetails</div>)
);

describe("Home Component", () => {
  it("renders all child components", () => {
    render(<Home />);

    expect(screen.getByText("Mocked SideBar")).toBeInTheDocument();
    expect(screen.getByText("Mocked BlogList")).toBeInTheDocument();
    expect(screen.getByText("Mocked BlogDetails")).toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
