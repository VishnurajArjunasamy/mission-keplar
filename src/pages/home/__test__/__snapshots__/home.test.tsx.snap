// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div>
      Mocked SideBar
    </div>
    <div>
      Mocked BlogList
    </div>
    <div>
      Mocked BlogDetails
    </div>
    <section
      aria-atomic="false"
      aria-label="Notifications Alt+T"
      aria-live="polite"
      aria-relevant="additions text"
      class="Toastify"
    />
  </div>
</DocumentFragment>
`;
